libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/auditing/restricted/AuditRestrictedCheckCoi.java

```diff
@@ -9,11 +9,27 @@
 import java.util.List;
 
 @ExcludeFromGeneratedCoverageReport
-public record AuditRestrictedCheckCoi(
-        @JsonInclude(JsonInclude.Include.NON_NULL) List<Name> oldName,
-        @JsonInclude(JsonInclude.Include.NON_NULL) List<Name> newName,
-        @JsonInclude(JsonInclude.Include.NON_NULL) List<BirthDate> oldBirthDate,
-        @JsonInclude(JsonInclude.Include.NON_NULL) List<BirthDate> newBirthDate,
-        @JsonProperty(value = "device_information", required = true)
-                DeviceInformation deviceInformation)
-        implements AuditRestricted {}
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class AuditRestrictedCheckCoi extends AuditRestrictedDeviceInformation {
```

Comment:
This it could be worth considering whether an interface would actually reduce code - consider:
```
interface AuditRestrictedWithDeviceInformation extends AuditRestricted {
    DeviceInformation deviceInformation();
}

public record AuditRestrictedCheckCoi(
    List<Name> oldName,
    List<Name> newName,
    List<BirthDate> oldBirthDate,
    List<BirthDate> newBirthDate,
    @JsonProperty("device_information")
    DeviceInformation deviceInformation)
  implements AuditRestrictedWithDeviceInformation {}

public record AuditRestrictedDeviceInformation(
    @JsonProperty("device_information")
    DeviceInformation deviceInformation
) implements AuditRestrictedWithDeviceInformation {
    public AuditRestrictedDeviceInformation(String deviceInformation) {
        this(new DeviceInformation(deviceInformation));
    }
}
```
But tbh I think it's always going to be a bit boilerplatey.